### PR TITLE
[otbn] Specify memory layout as seen from processor

### LINKED ISSUE #2620: [OTBN] Define memory map as seen from "inside" OTBN
Code executing on OTBN needs to address instruction and data memory. We have defined that this addressing is using byte addresses, but we have not yet defined the memory layout.

Taken from dcrypto is the assumption that imem and dmem is a strict Harvard architecture with imem and dmem starting both at address 0.

There has also been thinking about a data rom and/or instruction rom.

Goal of this issue: discuss the options, decide for one, and update the specification document with it.

> estimate 2

### PR DESCRIPTION
**Pull request by imphil**
Fixes #2620

### CODE REVIEW

**Comment by imphil** [comment]
We do have those points covered elsewhere, but since I've now added the memory part to the programmer's guide, it's probably good to re-iterate those points there as well. I'll update this PR.

File: hw/ip/otbn/doc/_index.md [hunk 1]
```diff
@@ -488,6 +488,11 @@ def StoreWlenWordToMemory(byteaddr: integer, storedata: Bits(WLEN)):
   This section will be written as we move on in the design and implementation process.
 </div>
 
+## Memory Layout
+
+The OTBN processor has access to dedicated and separate instruction and data memories, following the Harvard architecture.
+Both memories are 4 KB in size and addressed starting at 0.
```
Comment (GregAC) [inline hw/ip/otbn/doc/_index.md:494]: KiB maybe?

**Comment by imphil** [comment]
@GregAC @felixmiller I changed the text considerably to include additional information, please have another look.

File: hw/ip/otbn/doc/_index.md [hunk 2]
```diff
@@ -488,6 +488,22 @@ def StoreWlenWordToMemory(byteaddr: integer, storedata: Bits(WLEN)):
...
+
+The OTBN processor core has access to two dedicated memories:
+an instruction memory (IMEM), and a data memory (DMEM).
+Each memory is 4 kiB in size.
+
+The memory layout follows the Harvard architecture.
+Both memories are byte-addressed, with addresses starting at 0.
+
+The instruction memory (IMEM) is 32b wide and provides the instruction stream to the OTBN processor;
+it cannot be read or written from user code thorugh load or store instructions.
```
Comment (GregAC) [inline hw/ip/otbn/doc/_index.md:501]: thorugh -> through

**Pull request merged**